test/uk/gov/hmrc/helptosave/models/PayePersonalDetailsSpec.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosave.models
+
+import play.api.libs.json.Json
+import uk.gov.hmrc.helptosave.utils.{TestData, TestSupport}
+
+class PayePersonalDetailsSpec extends TestSupport with TestData { // scalastyle:off magic.number
```

Comment:
could we have unit tests to make sure the preference order of names and addresses is preserved as well?